### PR TITLE
Fix rename issue

### PR DESCRIPTION
**Pull request by shayzluf**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

rename (move) fails on some filesystems. this pr changes to use copy instead

### CODE REVIEW

File: shared/fileutil/fileutil.go [hunk 1]
```diff
@@ -78,3 +78,18 @@ func ReadFileAsBytes(filename string) ([]byte, error) {
...
 }
+
+// CopyFile copy a file from source to destination path.
+func CopyFile(src, dst string) error {
+	input, err := ioutil.ReadFile(src)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(dst, input, 0644)
```
Comment (nisdas) [inline shared/fileutil/fileutil.go:89]: use the file permissions from our config

**Review by nisdas — COMMENTED** [review]
Also can you add a basic sanity test for the copy function

File: shared/fileutil/fileutil.go (hunk 1 above)
Comment (shayzluf) [reply shared/fileutil/fileutil.go:89]: thanks

File: shared/fileutil/fileutil.go [hunk 2]
```diff
@@ -78,3 +80,18 @@ func ReadFileAsBytes(filename string) ([]byte, error) {
...
+// CopyFile copy a file from source to destination path.
+func CopyFile(src, dst string) error {
+	input, err := ioutil.ReadFile(src)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(dst, input, params.BeaconIoConfig().ReadWritePermissions)
+	if err != nil {
+		err := errors.Wrapf(err, "Error creating file: %s", dst)
```
Comment (prestonvanloon) [inline shared/fileutil/fileutil.go:93]:
```suggestion
		err := errors.Wrapf(err, "error creating file %s", dst)
```

File: shared/fileutil/fileutil_test.go [hunk 3]
```diff
@@ -13,13 +13,22 @@
...
+	"log"
 	"os"
 	"os/user"
 	"testing"
 
+	"github.com/prysmaticlabs/prysm/shared/params"
+
+	"github.com/prysmaticlabs/prysm/shared/fileutil"
+
+	"github.com/prysmaticlabs/prysm/shared/testutil"
```
Comment (prestonvanloon) [inline shared/fileutil/fileutil_test.go:31]:
```suggestion
	"github.com/prysmaticlabs/prysm/shared/params"
	"github.com/prysmaticlabs/prysm/shared/fileutil"
	"github.com/prysmaticlabs/prysm/shared/testutil"
```
You might need to run goimports to sort these

File: shared/fileutil/fileutil_test.go [hunk 4]
```diff
@@ -35,8 +44,48 @@ func TestPathExpansion(t *testing.T) {
...
+	df, err := os.Open(file2)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	sscan := bufio.NewScanner(sf)
+	dscan := bufio.NewScanner(df)
+
+	for sscan.Scan() {
+		dscan.Scan()
```
Comment (prestonvanloon) [inline shared/fileutil/fileutil_test.go:85]:
```suggestion
	for sscan.Scan() && dscan.Scan() {
```

File: shared/fileutil/fileutil_test.go [hunk 5]
```diff
@@ -35,8 +44,48 @@ func TestPathExpansion(t *testing.T) {
...
+		assert.NoError(t, os.Remove(fName+"copy"))
+	}()
+
+	assert.Equal(t, true, deepCompare(fName, fName+"copy"))
+}
+
+func deepCompare(file1, file2 string) bool {
+	sf, err := os.Open(file1)
+	if err != nil {
+		log.Fatal(err)
```
Comment (prestonvanloon) [inline shared/fileutil/fileutil_test.go:73]: Instead of using log.Fatal, please use t.Fatal

File: shared/fileutil/fileutil.go [hunk 6]
```diff
@@ -8,6 +8,8 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/prysmaticlabs/prysm/shared/params"
+
```
Comment (prestonvanloon) [inline shared/fileutil/fileutil.go:12]:
```suggestion
```

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**